modules/sdk-coin-sol/src/lib/transactionBuilderFactory.ts

```diff
@@ -29,8 +29,16 @@ export class TransactionBuilderFactory extends BaseTransactionBuilderFactory {
     try {
       switch (tx.type) {
         case TransactionType.Send:
-          if (tx.inputs[0].coin === 'sol' || tx.inputs[0].coin === 'tsol') {
-            return this.getTransferBuilder(tx);
+          const uniqueInputCoins = tx.inputs
+            .map((input) => input.coin)
+            .filter((coin, index, arr) => arr.indexOf(coin) === index);
+          if (uniqueInputCoins.includes('sol') || uniqueInputCoins.includes('tsol')) {
+            if (uniqueInputCoins.length > 1) {
+              // multi-assets transfer tx including native and token transfer
+              return this.getTransferBuilderV2(tx);
```

Comment:
Nit: can probably make this utilize a `?` operator, but for clarity this also seems fine.